geode-wan/src/distributedTest/java/org/apache/geode/management/internal/cli/commands/AlterRegionCommandWithRemoteLocator.java

```diff
@@ -207,8 +207,34 @@ public void whenAlteringMultipleRegionWithAlterCommandToAddGatewaySendersThenItS
           "alter region --name=SecurityCrossReferences --gateway-sender-id=serialSender1,serialSender2");
       return true;
     });
+  }
 
+  /**
+   * Test execution of alter region command, when adding gw sender to already initialized region is
+   * not taking too much time.
+   * Verify that execution of command will not be halted, due to PRShadowRegion not available
+   * immediately in all servers.
+   */
+  @Test
+  public void alterInitializedRegionWithGwSenderOnManyServersDoesNotTakeTooLong() {
+    gfsh.executeAndAssertThat("create disk-store --name=data --max-oplog-size=10 --dir=.")
+        .statusIsSuccess();
 
-  }
+    gfsh.executeAndAssertThat(
+        "create region --name=Positions --type=PARTITION_REDUNDANT_PERSISTENT --disk-store=data --total-num-buckets=1103")
+        .statusIsSuccess();
+
+    gfsh.executeAndAssertThat(
+        "query --query=\"select key,value from " + SEPARATOR + "Positions.entries\"")
+        .statusIsSuccess();
+
+    gfsh.executeAndAssertThat(
+        "create gateway-sender --id=parallelPositions --remote-distributed-system-id=1 --enable-persistence=true --disk-store-name=data --parallel=true")
+        .statusIsSuccess();
 
+    GeodeAwaitility.await().until(() -> {
```

Comment:
Shouldn't you add an atMost() condition here?

Reply:
In previos PR this was commented, and updated accordingly.

Reply:
I agree with you.